### PR TITLE
[wip] dev/core#3701 Fix import notices

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
E-notice on the match fields forms: Notice: Undefined index: isCheked in include() (line 35 of /srv/buildkit/build/dmaster/web/sites/default/files/civicrm/templates_c/en_US/%%72/72E/72E39105%%MapField.tpl.php)

Before
----------------------------------------
The above notice affects all imports

After
----------------------------------------
Notice fixed - see https://lab.civicrm.org/dev/core/-/issues/3701#note_77844 for discussion of the behaviour

Technical Details
----------------------------------------
This is a regression to the extent that it used to only affect one import and now affects many - I had thought to leave it out of scope for the current re-write but two people put up PRs to try to fix it so I figured I should do a more complete fix. This fix also consolidates duplicated code into these two functions - https://github.com/civicrm/civicrm-core/compare/master...eileenmcnaughton:civicrm-core:import_notice?expand=1#diff-f664874b9e78184d0a62439b5c070c7d2195ff19a5da2ce5c340483205aca00aR401 and most of the code changes are to that end.

This is a piece of string - I wound up questioning why we use js to make two checkboxes work like a radio button, should we re-organize the js, should we use an entity reference field, is the sky really blue - but at the end of the day this does include a good chunk of cleanup & addresses the issue

Comments
----------------------------------------
Review would be primarily UI testing

### CODE REVIEW

**Review by tschuettler — COMMENTED** [review]
Starting an import without selecting a mapping in the first step produces a fatal error:
> TypeError: CRM_Import_Forms::getMappingID(): Return value must be of type ?int, string returned in CRM_Import_Forms->getMappingID() (line 701 of /home/jenkins/bknix-dfl/build/core-23950-545vf/web/sites/all/modules/civicrm/CRM/Import/Forms.php).

File: templates/CRM/Contact/Import/Form/Preview.tpl [hunk 1]
```diff
@@ -142,14 +142,6 @@
...
 {literal}
 <script type="text/javascript">
-
-{/literal}{if $invalidGroupName}{literal}
-cj("#new-group.collapsed").crmAccordionToggle();
-{/literal}{/if}{literal}
-
-{/literal}{if $invalidTagName}{literal}
-cj("#new-tag.collapsed").crmAccordionToggle();
-{/literal}{/if}{literal}
```
[the remark this answers is not in the document]
Comment (eileenmcnaughton) [reply templates/CRM/Contact/Import/Form/Preview.tpl:152]: Yep :-)

File: CRM/Import/Form/MapField.php [hunk 2]
```diff
@@ -422,4 +393,70 @@ protected function addMapper(): array {
...
+        $mappingTypeId = CRM_Core_PseudoConstant::getKey('CRM_Core_BAO_Mapping', 'mapping_type_id', 'Import Contact');
+        if (CRM_Core_BAO_Mapping::checkMapping($nameField, $mappingTypeId)) {
+          $errors['saveMappingName'] = ts('Duplicate Import Mapping Name');
+        }
+      }
+    }
+    $template = CRM_Core_Smarty::singleton();
+    if (!empty($fields['saveMapping'])) {
+      $template->assign('isCheked', TRUE);
+    }
```
[the remark this answers is not in the document]
Comment (eileenmcnaughton) [reply CRM/Import/Form/MapField.php:458]: good point

**Comment by eileenmcnaughton** [comment]
@tschuettler I just pushed up a fix for that type error - hopefully its' OK now

**Comment by eileenmcnaughton** [comment]
I tried to get my head back into this but it was hard with such a lot of change - I pulled off a sub-PR here https://github.com/civicrm/civicrm-core/pull/24217 - which focusses only on 'safe' parts of 'activity' and 'contact' - but possibly addresses the type casting of `mappingId` for when the rest of this can be applied

**Comment by eileenmcnaughton** [comment]
Closing out for now - many parts merged but currently thoroughly stale.

Note that the current rc has a new UI in an extension for contribution imports

**Pull request closed without merging**